data/dataset/snowflake_example_test_dataset.yml

```diff
@@ -0,0 +1,229 @@
+dataset:
+  - fides_key: snowflake_example_test_dataset
+    name: Snowflake Example Test Dataset
+    description: Example of a Snowflake dataset containing a variety of related tables like customers, products, addresses, etc.
+    collections:
+      - name: address
+        fields:
+          - name: city
+            data_categories: [user.provided.identifiable.contact.city]
+          - name: house
+            data_categories: [user.provided.identifiable.contact.street]
+          - name: id
+            data_categories: [system.operations]
+            fidesops_meta:
+              primary_key: True
+          - name: state
+            data_categories: [user.provided.identifiable.contact.state]
+          - name: street
+            data_categories: [user.provided.identifiable.contact.street]
+          - name: zip
+            data_categories: [user.provided.identifiable.contact.postal_code]
+
+      - name: customer
+        fields:
+          - name: address_id
+            data_categories: [system.operations]
+            fidesops_meta:
+              references:
+                - dataset: snowflake_example_test_dataset
+                  field: address.id
+                  direction: to
+          - name: created
+            data_categories: [system.operations]
+          - name: email
+            data_categories: [user.provided.identifiable.contact.email]
+            fidesops_meta:
+              identity: email
+              data_type: string
+          - name: id
+            data_categories: [user.derived.identifiable.unique_id]
+            fidesops_meta:
+              primary_key: True
+          - name: name
+            data_categories: [user.provided.identifiable.name]
+            fidesops_meta:
+              data_type: string
+              length: 40
+          - name: variant_eg
+            # We use this data category so we can target this column from
+            # our Snowflake tests
+            data_categories: [user.provided.identifiable.name]
```

Comment:
This column maps to a Snowflake specific datatype that I've added to the schema — this is where having schemas defined in our application code would really help. as it would be obvious if that were the case. I don't think we have a `data_type` annotation for `variant` yet, so I didn't add it here, but that would have also been a clue.

The variant type is more or less a wildcard column, with some other minor nuances. https://docs.snowflake.com/en/sql-reference/data-types-semistructured.html#variant

Reply:
got it, thanks!